src/HaskellCodeExplorer/PackageInfo.hs

```diff
@@ -143,41 +148,33 @@ createPackageInfo packageDirectoryPath mbDistDirRelativePath sourceCodePreproces
           Right distDir -> return distDir
           Left errorMessage ->
             logErrorN (T.pack errorMessage) >> liftIO exitFailure
-  eitherPackageGhcVersion <- liftIO $ getPackageGhcVersion distDir
-  case eitherPackageGhcVersion of
-    Right packageGhcVersion ->
-      if take 2 (versionBranch packageGhcVersion) == take 2 (versionBranch ghcVersion)
-        then return ()
-        else let message =
-                   "GHC version mismatch. haskell-code-indexer: " ++
-                   showVersion ghcVersion ++
-                   ", package: " ++
-                   showVersion packageGhcVersion
-              in logErrorN (T.pack message) >> liftIO exitFailure
-    Left err -> logErrorN (T.pack err) >> liftIO exitFailure
-  let cabalHelperQueryEnv = mkQueryEnv packageDirectoryAbsPath distDir
-  ((packageName, packageVersion), compInfo) <-
-    liftIO $
-    runQuery
-      cabalHelperQueryEnv
-      ((,) <$> packageId <*>
-       (zip3 <$> components ((,) <$> ghcOptions) <*>
-        components ((,) <$> entrypoints) <*>
-        components ((,) <$> sourceDirs)))
-  let currentPackageId = HCE.PackageId (T.pack packageName) packageVersion
+  cabalHelperQueryEnv <- liftIO $ mkQueryEnv (ProjLocV1Dir packageDirectoryAbsPath) (DistDirV1 distDir)
+  ("ghc", packageGhcVersion) <- liftIO $ runQuery compilerVersion cabalHelperQueryEnv
+  unless (take 2 (versionBranch packageGhcVersion) == take 2 (versionBranch ghcVersion)) $
```

Comment:
It seem that the version of GHC from the header of `setup-config` contains only two numbers - major and minor (`showHeader` function from `Cabal` uses `compilerVersion` from System.Info: http://hackage.haskell.org/package/base-4.12.0.0/docs/src/System.Info.html#compilerVersion).

Reply:
Ah damn, I totally forgot. That won't do at all then, I'll have to revert that `compilerVersion` optimization then. I think there's just no way around actually going through the helper to get the GHC version then.

Honestly that shouldn't be too big a deal though since that will only happen once per Cabal version, regardless of which GHC version is compiling the helper. So the error will take a while longer to pop up but when the user gets to re-trying with a different GHC the helper will already be ready.

Reply:
OH! Even worse if the header version is from System.Info like you say then this is the GHC version that compiled `Setup.hs` and not the one `Setup.hs` decided to use to build the package! Yikes, that's just completely wrong then.

Reply:
Does it mean that the correct version of GHC should always be taken from `LocalBuildInfo`, not from the header of `setup-config`?